Debug operator (#5196)
brimdata/zed: docs/language/operators/debug.md

### Operator

&emsp; **debug** &mdash; write intermediate values to stderr

### Synopsis

```
debug [ <expr> ]
```
### Description

> The `debug` operator write the value of `expr` to the debug channel. If no
> `expr` is provided `this` is written to the debug channel. If the query is
> ran on command line via `zq` or `zed` all output written to the debug
> channel is displayed on stderr.
> 
> The `debug` operator is useful to view intermediate values as part of
> workflow when debugging complex zed query.

Reviewer: ```suggestion
The `debug` operator writes the value of `expr` to the debug channel. If no
`expr` is provided, `this` is written to the debug channel. If the query is
run on the command line via `zq` or `zed` all output written to the debug
channel is displayed on stderr.

The `debug` operator is useful to view intermediate values when debugging a
complex Zed query.
```

Author: Much better, thanks

### Examples

> The following query:
> ```
> echo '"foo"' | zq -z 'debug f"debug: {this}" | yield f"{this}_bar"' -
> ```
> will display `"debug: foo"` on stderr whereas `"foo_bar"` will display
> on stdout.

Reviewer: ````suggestion
The following query uses expressions containing [f-strings](../expressions.md#formatted-string-literals)
to display `"debug: foo"` on stderr whereas `"foo_bar"` will display
on stdout.

```
echo '"foo"' | zq -z 'debug f"debug: {this}" | yield f"{this}_bar"' -
```
````
